### ISSUE TITLE
TORUS rotation issue

### ISSUE THREAD

**Issue by ypeneliau** [arekfu/t4_geom_convert#34]
Dear development team,

I have noticed some issues when applying a rotation to a universe when this rotation modifies the axis of a torus. I have prepared some simplified models in which the issue is highlighted. I join the corresponding data files
- torusx.txt :  a simple torus with X axis
- torusx.t4.txt : a simple torus with X ax is conversion into TRIPOLI-4 data
- torusx.out.txt : the output of the conversion tool
- torus_rot.i.txt : a rotated torus with W axis, the rotation applying in XY plane
- torus_rot.t4.txt : a "voided" data file -> the conversion failed
- torus_rot.out.txt 
 : the output of the conversion tool
Thanks a lot for considering my request
Best regards
Yannick PENELIAU

[torusx.i.txt](https://github.com/arekfu/t4_geom_convert/files/6312843/torusx.i.txt)
[torusx.out.txt](https://github.com/arekfu/t4_geom_convert/files/6312844/torusx.out.txt)
[torusx.t4.txt](https://github.com/arekfu/t4_geom_convert/files/6312845/torusx.t4.txt)
[torusx_rot.i.txt](https://github.com/arekfu/t4_geom_convert/files/6312848/torusx_rot.i.txt)
[torusx_rot.out.txt](https://github.com/arekfu/t4_geom_convert/files/6312849/torusx_rot.out.txt)
[torusx_rot.t4.txt](https://github.com/arekfu/t4_geom_convert/files/6312850/torusx_rot.t4.txt)

**Comment by arekfu** [comment]
Hello Yannick, the [torusx_rot.i.txt](https://github.com/arekfu/t4_geom_convert/files/6312848/torusx_rot.i.txt) file actually contains a small syntax error. The coordinate transformation, as far as I am aware, should read
```
*tr1  0 10 10   30 60 90   120 30 90   90 90 0
```
Note the `*` at the beginning of the line. With this correction, `t4_geom_convert` chokes on the conversion with the "right" error message:
```
t4_geom_convert.Kernel.Surface.SurfaceConversionError.SurfaceConversionError: Cannot convert TORUS with generic axis: (0.8660254037844387, 0.5000000000000001, 6.123233995736766e-17)
```
I'll try to add support for generic tori via T4's volume `TYPE TORUS`. However, I have noticed that this can only handle circular tori. Generically aligned elliptical tori are not supported in any form by T4 at the moment. Does your real use case involve circular or elliptical tori?

**Comment by arekfu** [comment]
So it looks that the `tr1*` syntax is actually allowed in MCNP. Support for this syntax in `t4_geom_convert` is tracked in issue #35.

**Comment by arekfu** [comment]
This issue was fixed with the release of v0.6.0.
